src/ex4/dist/main.js

```diff
@@ -1,52 +1,194 @@
+/* eslint-disable camelcase */
+import DomManager from './services/DomManager.js';
+import {
+  GetResourceRequest, AddNewResourceRequest, DeleteResourceRequest, PatchResourceRequest, PutResourceRequest
+} from './clients/ApiRequest.mjs';
+// Implement the `Main` class here
+
 class Main {
-    constructor() {
-        this.itemClient = new ItemClient()
-    }
+  constructor() {
+    this.dom_manager = new DomManager();
+    this.pokemons = [];
+  }
 
-    init = async () => {
-        const addItemButton = document.getElementById("list-item-submit");
-        addItemButton.addEventListener("click", this.handleItem);
+  /**
+     * init all event listeners and image animation
+     */
+  async init(pokemons) {
+    this.pokemons = pokemons;
 
-        await this.renderItems(); // this will make it so that any time you refresh the page you'll see the items already in your todo list
-    }
+    this.dom_manager.add_task_button.addEventListener('click', () => {
+      this.AddButtonCallBack();
+    });
 
-    handleItem = async () => {
-        // implement
-    }
+    // on enter key press event
+    this.dom_manager.task_input.addEventListener('keypress', (event) => {
+      this.dom_manager.AddNewTaskByKeyPress(event);
+    });
 
-    deleteItem = async item => {
-        // implement
-    }
+    this.dom_manager.clear_all_button.addEventListener('click', () => {
+      this.ClearAllTasks();
+    });
+
+    this.dom_manager.sort_by_name_button.addEventListener('click', () => {
+      this.SortTasksByName();
+    });
 
-    renderItems = async () => {
-        const list = document.getElementById("list");
-        list.innerHTML = "";
+    setInterval(this.dom_manager.PokemonImageAnimation, 500);
 
-        const items = 'where do you get the items from now that you have a server..?'
+    this.dom_manager.AddPokemonsToDataList(this.pokemons);
+  }
 
-        items.forEach(item => {
-            const listItem = document.createElement("li");
-            listItem.classList.add('list-item');
-            listItem.innerHTML = item;
+  /**
+     * add button call back function
+     */
+  async AddButtonCallBack() {
+    const task_text_from_user = this.dom_manager.task_input.value;    
+    // check if input is not empty
+    if (task_text_from_user) {
```

Comment:
It is a best practice to check if the `task_text_from_user` is different from an empty string. You could also trim the value.